### ISSUE TITLE
Compiling 3.7.0 won't work until base-includes.h is replaced with extension.h

### ISSUE THREAD

**Issue by ckortekaas** [chregu/hhvm-newrelic-ext#27]
Hi @chregu 

I've tried compiling HHVM 3.7 against 0.16.1 (see my Docker build script here https://github.com/ckortekaas/hhvm-nginx-newrelic-docker/blob/master/Dockerfile) and it fails with:
```
 [ 50%] Building CXX object CMakeFiles/newrelic.dir/newrelic.cpp.o
 /usr/local/src/hhvm-newrelic-ext/newrelic.cpp:1:45: fatal error: hphp/runtime/base/base-includes.h: No   such file or directory
``` 
After a bit of searching I found that in 3.7 they removed base-includes.h from HHVM: https://github.com/facebook/hhvm/commit/91e07b951496b5bfbbb189fd5401e9dd5e711c43

Apparently they want you to use #include "hphp/runtime/ext/extension.h" instead.

Cheers,
Christiaan Kortekaas

**Comment by ckortekaas** [comment]
To follow up, after making that change it fails next at:

```
[ 50%] Building CXX object CMakeFiles/newrelic.dir/newrelic.cpp.o
In file included from /usr/local/include/hphp/runtime/ext/extension.h:23:0,
                 from /usr/local/src/hhvm-newrelic-ext/newrelic.cpp:1:
/usr/local/include/hphp/runtime/vm/native.h: In instantiation of 'void HPHP::Native::registerBuiltinFunction(const char*, Fun) [with Fun = long int (*)(const HPHP::String&, const HPHP::String&, const HPHP::String&, const HPHP::String&, HPHP::String&)]':
/usr/local/src/hhvm-newrelic-ext/newrelic.cpp:251:9:   required from here
/usr/local/include/hphp/runtime/vm/native.h:368:3: error: static assertion failed: Arguments on builtin function were not understood types
   static_assert(
   ^
```

and 

```
/usr/local/include/hphp/runtime/vm/native.h: In instantiation of 'void HPHP::Native::registerBuiltinFunction(const char*, Fun) [with Fun = HPHP::Variant (*)(const HPHP::String&, const HPHP::String&, HPHP::String, HPHP::String)]':
/usr/local/src/hhvm-newrelic-ext/newrelic.cpp:255:9:   required from here
/usr/local/include/hphp/runtime/vm/native.h:368:3: error: static assertion failed: Arguments on builtin function were not understood types
```

Any ideas what to do next to fix this?

**Comment by fabianhenzler** [comment]
hhvm-newrelic-ext git:(master) ✗ make
-- Configuring for HHVM API version 20150212
-- Configuring done
-- Generating done
-- Build files have been written to: /root/newrelic/hhvm-newrelic-ext
[ 50%] Building CXX object CMakeFiles/newrelic.dir/newrelic.cpp.o
/root/newrelic/hhvm-newrelic-ext/newrelic.cpp:1:45: fatal error: hphp/runtime/base/base-includes.h: No such file or directory
 #include "hphp/runtime/base/base-includes.h"
                                             ^
compilation terminated.
make[2]: *** [CMakeFiles/newrelic.dir/newrelic.cpp.o] Error 1
make[1]: *** [CMakeFiles/newrelic.dir/all] Error 2
make: *** [all] Error 2

**Comment by fabiorphp** [comment]
Please see this pull request #28

**Comment by ckortekaas** [comment]
Thanks @fabiorphp - I've got a working docker container for new relic and hhvm 3.7 using your help here: https://github.com/ckortekaas/hhvm-nginx-newrelic-docker and https://registry.hub.docker.com/u/ckortekaas/hhvm-nginx-newrelic/

**Comment by chregu** [comment]
It's merged and I updated the README to mention 3.7.0

Thanks for all the help
